Fix some minor issues in FSharp docs (#33730)
dotnet/docs: docs/fsharp/style-guide/formatting.md

if a then ()
```

> If any of the expressions are multi-line, each conditional branch should be multi-line.

Author: I wasn't sure if I should update this for both of these cases, since they're essentially describing the same thing, one just includes some `elif` branches, so I aired on the side of including both.

```fsharp
// ✔️ OK
if cond then

>     let e1 = something()

Author: Both of these sections were describing how to format expressions when the content was multi-line, but neither actually had a multi-line expression in them, so that's why I added these `let e1 = something()` lines.

I wasn't sure if I should update this for both of these cases, since they're essentially describing the same thing, one just includes some `elif` branches, but I aired on the side of including both for completeness' sake.

### Formatting record declarations

> For record declarations, by default you should indent `{` in type definitions by four spaces, start the field list on the same line, and align any members with the `{` token:

Reviewer: ```suggestion
For record declarations, by default you should indent the `{` in the type definition by four spaces, start the label list on the same line, and align members, if any, with the `{` token:
```

```fsharp

> // ✔️ OK

Reviewer: Should line 1696 say "...with **labels** indented by an additional four spaces"?

Author: No, it's referring to additional members as described [here](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/records#records-and-members). I'll update the example to include one for clarity

    }
```

> You may also put the `{` at the end of the type declaration line like (`Stroustrup` style):

Reviewer: ```suggestion
You can also put the `{` at the end of the first line of the type definition (`Stroustrup` style):
```
